sql/sys_vars.h

```diff
@@ -416,6 +416,10 @@ class Sys_var_enum : public Sys_var_typelib {
   const uchar *global_value_ptr(THD *, LEX_STRING *) {
     return pointer_cast<const uchar *>(typelib.type_names[global_var(ulong)]);
   }
+  void persist_only_update(THD *, set_var *var, char *dest) {
+    strcpy(dest,
```

Comment:
Multiple occurrences: strcpy does not prevent buffer overflow. use strncpy instead.

Reply:
This part was changed to use String.set()